apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/DisplayServiceAccountCard.tsx

```diff
@@ -0,0 +1,241 @@
+        </Stack>
+      );
+    } else if (adminApiError) {
+      return (
+        <RenderSimpleBadgeNote badgeLabel="Admin API Error" noteMessage={adminApiError.message} />
+      );
+    } else if (dataApiError) {
+      return (
+        <RenderSimpleBadgeNote badgeLabel="Data API Error" noteMessage={dataApiError.message} />
+      );
+    } else if (unknownError) {
+      return (
+        <RenderSimpleBadgeNote badgeLabel="Unknown Error" noteMessage={unknownError.message} />
+      );
+    }
+
+    return <Badge variant="positive">Active</Badge>;
+  };
+
+  return (
+    <Card>
+      <Flex justifyContent="space-between" marginBottom="none">
+        <Paragraph marginBottom="none" marginTop="spacingXs">
+          <b>Google Service Account Details</b>
+        </Paragraph>
+        <Flex justifyContent="space-between" marginBottom="spacingL">
+          <Box paddingRight="spacingXs" paddingTop="spacingXs">
+            <TextLink
+              testId="editServiceAccountButton"
+              as="button"
+              variant="primary"
+              onClick={onEditGoogleAccountDetails}>
+              Edit
+            </TextLink>
+          </Box>
+          <Box style={{ minWidth: '60px', minHeight: '30px' }}>
+            {isLoading ? (
+              <Spinner variant="default" />
+            ) : (
+              <Button variant="primary" size="small" onClick={handleApiTestClick}>
+                Test
+              </Button>
+            )}
+          </Box>
+        </Flex>
+      </Flex>
+      <FormControl>
+        <FormControl.Label marginBottom="none">Service Account</FormControl.Label>
+        <Paragraph>
+          <Flex alignItems="center">
+            <Box paddingRight="spacingS">
+              <TextLink
+                icon={<ExternalLinkTrimmedIcon />}
+                alignIcon="end"
+                href={`https://console.cloud.google.com/iam-admin/serviceaccounts/details/${serviceAccountKeyId.clientId}?project=${serviceAccountKeyId.projectId}`}
+                target="_blank"
+                rel="noopener noreferrer">
+                {serviceAccountKeyId.clientEmail}
+              </TextLink>
+            </Box>
+            {isLoading ? (
+              <Spinner variant="default" />
+            ) : adminApiError ? (
+              <ErrorCircleIcon variant="negative" />
+            ) : (
+              <CheckCircleIcon variant="positive" />
+            )}
+          </Flex>
+        </Paragraph>
+      </FormControl>
+      <FormControl>
+        <FormControl.Label marginBottom="none">Key ID</FormControl.Label>
+        <Paragraph>
+          <Box as="code">{serviceAccountKeyId.id}</Box>
+        </Paragraph>
+      </FormControl>
+      <FormControl marginBottom="none">
+        <FormControl.Label marginBottom="none">Status</FormControl.Label>
+        <Paragraph>{isLoading ? <Spinner variant="default" /> : <RenderStatusInfo />}</Paragraph>
+      </FormControl>
+    </Card>
+  );
+};
```

Comment:
this file overall is looking good, though I think it is quite large with a lot of logic and responsibilities to all be in one file and component. Not necessary to change now,  but what are your thoughts on extracting some of this into either smaller components, hooks, helpers etc?

Reply:
I don't see why not!